clap_generate/src/man.rs

```diff
@@ -0,0 +1,287 @@
+use std::fmt::write;
+
+use clap::ArgSettings;
+
+/// Man page generator
+#[derive(Debug, Clone)]
+pub struct Man<'a> {
+    app: clap::App<'a>,
+    section: i8,
+    manual: Option<String>,
+    sections: Vec<(String, Vec<String>)>,
+}
+
+impl<'a> Man<'a> {
+    /// Create a new man page for the app.
+    pub fn create(
+        app: clap::App<'a>,
+        section: i8,
+        manual: Option<String>,
+        buf: &mut dyn std::io::Write,
+    ) {
+        let man = Man {
+            app,
+            section,
+            manual,
+            sections: Vec::new(),
+        };
+
+        man.write(buf)
+    }
+
+    /// Create a new builder for man pages.
+    pub fn new(app: clap::App<'a>, section: i8, manual: Option<String>) -> Self {
+        Man {
+            app,
+            section,
+            manual,
+            sections: Vec::new(),
+        }
```

Comment:
So we're accepting an `app` by ownership.  

This is different than the shell generators which accept a `&mut App`.  I think I'm fine with people cloning their `App` if needed, its not like this will be in a hot loop.

Probably the most important thing is calling `app._build_all();` to ensure we've massaged things based on the users settings.

Reply:
Good point about `app._build_all()`, I fixed this :+1: